### PR TITLE
Presentation: Review breaking changes & cleanup changelogs

### PR DESCRIPTION
**Pull request by grigasp**
Closes https://github.com/iTwin/presentation/issues/347

### CODE REVIEW

File: .changeset/rude-years-allow.md [hunk 1]
```diff
@@ -1,5 +1,63 @@
...
+- `useHierarchyLevelFiltering` hook's result now contains functions `applyFilter` and `clearFilter` that take node identifier argument rather than a `TreeNodeItem`. The change was made to help reacting to the above `PresentationTreeNodeRenderer` change by requiring the same types of arguments as what `onClearFilterClick` and `onFilterClick` get. In case these functions are used outside of `PresentationTreeNodeRenderer` workflows, reacting to the breaking change is as follows:
+
+  *before*
+
+  ```tsx
+  const { applyFilter } = useHierarchyLevelFiltering({ nodeLoader, modelSource });
+  const [filterNode, setFilterNode] = useState<PresentationTreeNodeItem>();
+  const onFilterButtonClick = () => {
+    if (filterNode) {
+      applyFilter(filterNode);
```
Comment (saskliutas) [inline .changeset/rude-years-allow.md:48]:
```suggestion
  const onFilterChanged = (filter:  PresentationInstanceFilterInfo) => {
    if (filterNode) {
      applyFilter(filterNode, filter);
```

File: .changeset/rude-years-allow.md [hunk 2]
```diff
@@ -1,5 +1,63 @@
...
+  ```
+
+  *after*
+
+  ```tsx
+  const { applyFilter } = useHierarchyLevelFiltering({ nodeLoader, modelSource });
+  const [filterNodeId, setFilterNodeId] = useState<string>();
+  const onFilterButtonClick = () => {
+    if (filterNodeId) {
+      applyFilter(filterNodeId);
```
Comment (saskliutas) [inline .changeset/rude-years-allow.md:60]:
```suggestion
  const onFilterChanged = (filter:  PresentationInstanceFilterInfo) => {
    if (filterNodeId) {
      applyFilter(filterNodeId, filter);
```

File: .changeset/rude-years-allow.md [hunk 3]
```diff
@@ -1,5 +1,63 @@
...
 
-Add interactive and more detailed informational messages in tree and instance filter components.
+Tree: Added interactive and more detailed informational messages in the tree and its hierarchy level filtering components:
+
+- When a hierarchy level size exceeds given limit, a message is displayed, suggesting the results should be filtered to reduce the result set.
+- The hierarchy level filtering dialog informs whether provided filters reduce the result set to a small enough size to be displayed in the tree.
+
+Includes 2 breaking `@beta` API changes:
+
+- `PresentationTreeNodeRenderer` now takes `onClearFilterClick` and `onFilterClick` callback props with node identifier argument rather than `PresentationTreeNodeItem`. This was a necessary change to allow opening filtering dialog for a parent node from its child node. To react to this breaking change:
```
Comment (saskliutas) [inline .changeset/rude-years-allow.md:12]: I think this breaking change should be placed after `useHierarchyLevelFiltering`. In that case it's easier to react to `PresentationTreeNodeRenderer` breaking changes. Or we could provide different examples that are not based on `useHierarchyLevelFiltering`:

Before
```tsx
const [clickedNode, setClickedNode] = useState<PresentationTreeNodeItem>();
return <PresentationTreeNodeRenderer
  {...nodeProps}
  onFilterClick={(node: PresentationTreeNodeItem) => {
    setClickedNode(node);
  }}
/>;
```

After
```tsx
const [clickedNode, setClickedNode] = useState<PresentationTreeNodeItem>();
return <PresentationTreeNodeRenderer
  {...nodeProps}
  onFilterClick={(nodeId: string) => {
    const node = modelSource.getModel().getNode(nodeId);
    if (isTreeModelNode(node) && isPresentationTreeNodeItem(node.item)) {
      setClickedNode(node.item);
    }
  }}
/>;
```

**Review by saskliutas — APPROVED** [review]

File: .changeset/rude-years-allow.md (hunk 3 above)
Comment (grigasp) [reply .changeset/rude-years-allow.md:12]: applied the suggestion

**Pull request merged**